frontend/src/components/reports/routine/ActivityReportByTest.js

```diff
@@ -0,0 +1,183 @@
+import React, { useEffect, useState, useRef } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+import { Form, FormLabel, Grid, Column, Section, Button, Loading } from "@carbon/react";
+import CustomDatePicker from "../../common/CustomDatePicker";
+import { AlertDialog } from "../../common/CustomNotification";
+import config from "../../../config.json";
+import TestSelectForm from "../../workplan/TestSelectForm";
+import PanelSelectForm from "../../workplan/PanelSelectForm";
+import TestSectionSelectForm from "../../workplan/TestSectionSelectForm";
+import "../../Style.css";
+import { getFromOpenElisServer } from "../../utils/Utils";
+
+const ActivityReport = () => {
+  const intl = useIntl();
+  const mounted = useRef(false);
+  const [loading, setLoading] = useState(false);
+  const [notificationVisible, setNotificationVisible] = useState(false);
+  const [reportFormValues, setReportFormValues] = useState({
+    startDate: null,
+    endDate: null
+  });
+  const [testList, setTestList] = useState([]);
+  const [panelList, setPanelList] = useState([]);
+  const [unitList, setUnitList] = useState([]);
+
+  const handleChangeDatePicker = (datePicker, date) => {
+    setReportFormValues(prevState => ({
+      ...prevState,
+      [datePicker]: date
+    }));
+  };
+
+  const handleSubmit = async (reportType) => {
+    setLoading(true);
+    try {
+      const baseParams = getBaseParams(reportType);
+      const url = `${config.serverBaseUrl}/ReportPrint?${baseParams}`;
+      window.open(url, "_blank");
+      setNotificationVisible(true);
+    } catch (error) {
+      console.error("Error generating report:", error);
+    }
+    setLoading(false);
+  };
+  
+  const handleSelectedValue = (v, l) => {
+    if (mounted.current) {
+      setSelectedValue(v);
+    }
+  };
+
+  const handleSetValue = (v) => {
+    if (mounted.current) {
+      setSetValue(v);
+    }
+  };
+  const getBaseParams = (reportType) => {
+    const { startDate, endDate } = reportFormValues;
+    const formattedStartDate = startDate ? formatDate(startDate) : '';
+    const formattedEndDate = endDate ? formatDate(endDate) : '';
+    let baseParams = '';
+    switch (reportType) {
+      case "test":
+        baseParams = `Routinereport=activityReportByTest&type=indicator&lowerDateRange=${formattedStartDate}&upperDateRange=${formattedEndDate}`;
```

Comment:
```suggestion
        baseParams = `RoutineReport?report=activityReportByTest&type=indicator&lowerDateRange=${formattedStartDate}&upperDateRange=${formattedEndDate}`;
```

Reply:
Thanks @manishjha-04   ohh i definately understand. this was @mozzy11  idea. i can definately revert it  back.